services/provider/server/storageclassrequest.go

```diff
@@ -32,24 +32,32 @@ func newStorageClassRequestManager(cl client.Client, namespace string) (*storage
 	}, nil
 }
 
-// getStorageClassRequestName generates a name for a StorageClassRequest resource.
-func getStorageClassRequestName(consumerUUID, storageClassRequestName string) string {
+// getStorageClassRequestHash generates a hash for a StorageClassRequest based
+// on the MD5 hash of the StorageClassClaim name and storageConsumer UUID.
+func getStorageClassRequestHash(consumerUUID, storageClassClaimName string) string {
 	var s struct {
-		StorageConsumerUUID     string `json:"storageConsumerUUID"`
-		StorageClassRequestName string `json:"storageClassRequestName"`
+		StorageConsumerUUID string `json:"storageConsumerUUID"`
+
+		// This field was named before a small refactor to clarify
+		// what it actually represents, but is being kept as-is so as
+		// to not change the generative logic.
+		StorageClassClaimName string `json:"storageClassRequestName"`
```

Comment:
Shouldn't this be called `StorageClaimName`? we moved away from StorageClassClaim API and replaced it with StorageClaim API

Reply:
I didn't think it was worth the effort to unify that *everywhere* in the codebase on the provider side. That's another follow-up PR, if anything.
![image](https://github.com/red-hat-storage/ocs-operator/assets/1234462/a22821fd-6ae1-49ba-9988-f2a6e80708bd)

Reply:
As the final results are changes we don't need to keep `storageClassRequestName` in the json we can just change it to `storageclaimName`. WDYT?